CredentialProvider.Microsoft/CredentialProviders/Vsts/MSAL/MsalTokenProvider.cs

```diff
@@ -43,10 +50,13 @@ private async Task<MsalCacheHelper> GetMsalCacheHelperAsync()
             // for now only support windows
             if (helper == null && this.cacheEnabled && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
+                this.Logger.Verbose($"Using MSAL cache at `{cacheLocation}`.");
+
                 var fileName = Path.GetFileName(cacheLocation);
                 var directory = Path.GetDirectoryName(cacheLocation);
 
-                var builder = new StorageCreationPropertiesBuilder(fileName, directory, this.clientId);
+                var builder = new StorageCreationPropertiesBuilder(fileName, directory);
+                builder = builder.WithCacheChangedEvent(this.clientId, "https://login.microsoftonline.com/common");
```

Comment:
Is this really needed? This allows you to subscribe to an event which tells you if the cache file has changed. I do not see an event handler for this though.

Reply:
We do not listen for cache change events in AzureAuth. I thought this was designed for long-running applications like VS.

Reply:
Removed